app/components/combined_system_message/index.js

```diff
@@ -3,21 +3,61 @@
 
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
+import {createSelector} from 'reselect';
 
 import {getProfilesByIds, getProfilesByUsernames} from 'mattermost-redux/actions/users';
 import {Preferences} from 'mattermost-redux/constants';
 import {getBool, getTeammateNameDisplaySetting} from 'mattermost-redux/selectors/entities/preferences';
-import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {getCurrentUser, getProfiles} from 'mattermost-redux/selectors/entities/users';
 
 import CombinedSystemMessage from './combined_system_message';
 
-function mapStateToProps(state) {
+const getUserProfiles = createSelector(
```

Comment:
Since this selector has props, this should be done as a selector factory like
```javascript
function makeGetUserProfiles() {
    return createSelector(...);
}

function makeMapStateToProps() {
    const getUserProfiles = makeGetUserProfiles();

    return (state, ownProps) => {...};
}
```